Source/WebCore/animation/KeyframeEffect.cpp

```diff
@@ -1849,7 +1849,8 @@ void KeyframeEffect::animationDidChangeTimingProperties()
 {
     computeSomeKeyframesUseStepsOrLinearTimingFunctionWithPoints();
     updateAcceleratedAnimationIfNecessary();
-    invalidate();
+    if (auto target = targetStyleable())
+        target->element.invalidateStyleInternal();
```

Comment:
Why does that not use `invalidateStyleForAnimation()` like `invalidate()` does?

Reply:
No reason!

Reply:
Turns out it was fixing something but the right place for the full invalidation is in KeyframeEffect::setKeyframes. New keyframes may change override interaction with the base style.